Bring profiles into the modern day

aarongable opened the pull request:
Overhaul Section 7.1 Profiles, with two goals in mind:
- Removing deprecated pieces of our profiles, such as OCSP and the `Internet Security Research Group` Organization Name.
- Giving them the same format as the profiles in Section 7.1 of the Baseline Requirements.

Fixes https://github.com/letsencrypt/cp-cps/issues/188
Fixes https://github.com/letsencrypt/cp-cps/issues/304

aarongable committed:
Bring profiles into the modern day

Overhaul Section 7.1 Profiles, with two goals in mind:
- Removing deprecated pieces of our profiles, such as OCSP and the `Internet Security Research Group` Organization Name.
- Giving them the same format as the profiles in Section 7.1 of the Baseline Requirements.

Fixes https://github.com/letsencrypt/cp-cps/issues/188
Fixes https://github.com/letsencrypt/cp-cps/issues/304

aarongable commented:
Preview the result here: https://github.com/letsencrypt/cp-cps/blob/aarongable-patch-1/CP-CPS.md#7-certificate-crl-and-ocsp-profiles

jsha commented:
Diffing the Root CA Certificate section against the corresponding section in the BRs:

 - For `version`, the BRs say "MUST be v3(2)" in the table; this branch refers out to a section.
 - `issuer`: the branch carries over "meaningful CN", but that's not in the BRs. I'd recommend deleting it. Also, the branch says just "ISRG" but some pre-existing roots still spell out "Internet Security Research Group"; I think it makes sense to include both here.
  - `validity`: the BRs refer out to a section; we inline it. Since a goal of this is closely matching the BRs, probably better to put our rules in a section to.
   - `subject`: the language here says it's byte-for-byte identical to `issuer`, but the BRs have it the other way around: `issuer` is described as byte-for-byte identical to `subject`, and `subject` has the naming description.
   - `subjectKeyIdentifier` - I think this is calculated differently for our old vs new roots and we should include both methods here.

aarongable committed:
Review comments

- Flip the statments about Root CA issuer and subject.
- Replace "meaningful" with "unique" when describing subject CN.

aarongable committed:
Rebase fixes

aarongable committed:
Edits for clarity

mcpherrinm commented on CP-CPS.md:
```diff
@@ -987,70 +987,116 @@ See [Section 5.5.5](#555-requirements-for-time-stamping-of-records).
+|     `serialNumber`                   | Approximately 128 bits, including at least 64 bits of output from a CSPRNG |
+|     `signature`                      | See [Section 7.1.3.2](#7132-signature-algorithmidentifier) |
+|     `issuer`                         | Byte-for-byte identical to the `subject` field of the Issuing CA |
+|     `validity`                       | At most 8 years |
```
we define root & leaf in terms of days now; should we do the same for cross & intermediate?

aarongable committed:
Express all validities in days

aarongable committed:
Change how we talk about serial numbers

Be more precise regarding randomness, and less precise regarding final length.

aarongable committed:
Give ourselves a buffer in both directions for serial length

aarongable committed:
Make keyEncipherment optional even for RSA keys

aarongable committed:
Clarify that cross-sign keys must match original keys
